pkg/statistics/handle/storage/read_test.go

```diff
@@ -116,3 +121,50 @@ func TestReloadExtStatsLockRelease(t *testing.T) {
 	tk.MustExec("alter table t add stats_extended s1 correlation(a,b)")
 	tk.MustExec("analyze table t") // no error
 }
+
+func TestLoadNonExistentIndexStats(t *testing.T) {
+	store, dom := testkit.CreateMockStoreAndDomain(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test;")
+	tk.MustExec("drop table if exists t;")
+	// Create table with an index. The index histogram will exist in the stats cache
+	// but won't have actual histogram data loaded yet.
+	tk.MustExec("create table if not exists t(a int, b int, index ia(a));")
+	tk.MustExec("insert into t value(1,1), (2,2);")
+	h := dom.StatsHandle()
+	require.NoError(t, h.DumpStatsDeltaToKV(true))
+	require.NoError(t, h.Update(dom.InfoSchema()))
+	// Trigger async load of index histogram by using the index in a query.
+	tk.MustExec("set tidb_opt_objective='determinate';")
```

Comment:
When we set this variable to true, it causes the physical table to no longer use a fake physical table ID, which in turn triggers statistics loading.
```go
func (idx *Index) CheckStats() {
	// When we are using stats from PseudoTable(), all column/index ID will be -1.
	if idx.IsFullLoad() || idx.PhysicalID <= 0 {
		return
	}
	HistogramNeededItems.Insert(model.TableItemID{TableID: idx.PhysicalID, ID: idx.Info.ID, IsIndex: true})
}

```